runtime/src/iree-amd-aie/aie_runtime/iree_aie_router.h

```diff
@@ -26,6 +27,12 @@ struct Port {
   typedef std::tuple<StrmSwPortType, int> TupleType;
   Port(TupleType t) : Port(std::get<0>(t), std::get<1>(t)) {}
   operator TupleType() const { return {bundle, channel}; }
+  friend llvm::hash_code hash_value(const Port &p) {
+    std::size_t h1 =
+        std::hash<mlir::iree_compiler::AMDAIE::StrmSwPortType>{}(p.bundle);
+    std::size_t h2 = std::hash<int>{}(p.channel);
+    return llvm::hash_combine(h1, h2);
+  }
```

Comment:
um what am i missing? you added this friend but you also saw there's already a std::hash functor (since you updated it to duplicate this).

Reply:
It's needed for `hash_combine` here: https://github.com/nod-ai/iree-amd-aie/pull/797/files#diff-f16ad809ac104ee8b9cc1d5d5f634f5a5cc35cf2396f29d261181ba4d30d9956R40